ts/Series/OnSeriesComposition.ts

```diff
@@ -183,17 +184,97 @@ namespace OnSeriesComposition {
                                 rightPoint &&
                                 typeof rightPoint[onKey] !== 'undefined'
                             ) {
-                                // the distance ratio, between 0 and 1
-                                distanceRatio =
-                                    ((point.x as any) - leftPoint.x) /
-                                    (rightPoint.x - leftPoint.x);
-                                (point.plotY as any) +=
-                                    distanceRatio *
-                                    // the plotY distance
-                                    (rightPoint[onKey] - leftPoint[onKey]);
-                                (point.y as any) +=
-                                    distanceRatio *
-                                    (rightPoint.y - leftPoint.y);
+                                // If the series is spline, calculate Y of the
+                                // point on the bezier line. #19264
+                                if (
+                                    defined(point.plotX) && (
+                                        onSeries.type === 'spline' ||
+                                        onSeries.type === 'areaspline'
+                                    )
+                                ) {
+                                    leftPoint = leftPoint as SplinePoint;
+                                    rightPoint = rightPoint as SplinePoint;
+
+                                    const p0 = {
+                                            plotX: leftPoint.plotX || 0,
+                                            plotY: leftPoint.plotY || 0
+                                        },
+                                        p3 = {
+                                            plotX: rightPoint.plotX || 0,
+                                            plotY: rightPoint.plotY || 0
+                                        },
+                                        p1 = leftPoint.rightCont || p0,
+                                        p2 = rightPoint.leftCont || p3,
+                                        pixelThreshold = 0.25,
+                                        maxIterations = 100,
+                                        calculateCoord = (
+                                            t: number,
+                                            key: 'plotX'|'plotY'
+                                        ): number | null => {
+                                            if (t < 0 || t > 1) {
+                                                return null;
+                                            }
+                                            return Math.pow(1 - t, 3) *
```

Comment:
I would add a reference or short description that's the cubic bezier explicit form equation